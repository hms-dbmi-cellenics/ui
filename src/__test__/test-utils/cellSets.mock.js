const mockCellSets = {
  initialLoadPending: false,
  error: false,
  loading: false,
  accessible: true,
  hierarchy: [
    {
      key: 'louvain',
      children: [{ key: 'cluster-a' }, { key: 'cluster-b' }, { key: 'cluster-c' }],
      cellIds: new Set(),
    },
    {
      key: 'sample',
      children: [{ key: 'sample-1' }, { key: 'sample-2' }],
      cellIds: new Set(),
    },
    {
      key: 'scratchpad',
      children: [{ key: 'scratchpad-a' }],
      cellIds: new Set(),
    },
  ],
  properties: {
    'cluster-a': {
      name: 'cluster a',
      key: 'cluster-a',
      cellIds: new Set([0, 1, 6]),
      color: '#01FFFF',
    },
    'cluster-b': {
      name: 'cluster b',
      key: 'cluster-b',
      cellIds: new Set([2, 3]),
      color: '#23FFFF',
    },
    'cluster-c': {
      name: 'cluster c',
      key: 'cluster-c',
      cellIds: new Set([4, 5]),
      color: '#45FFFF',
    },
    'sample-1': {
      name: 'Sample 1',
      key: 'sample-1',
      cellIds: new Set([0, 1, 2]),
      color: '#012FFF',
    },
    'sample-2': {
      name: 'Sample 2',
      key: 'sample-2',
      cellIds: new Set([3, 4, 5]),
      color: '#345FFF',
    },
    'scratchpad-a': {
      cellIds: new Set(['5']),
      key: 'scratchpad-a',
      name: 'New Cluster',
      color: '#5FFFFF',
    },
    louvain: {
      cellIds: new Set(),
      name: 'Louvain clusters',
      key: 'louvain',
      type: 'cellSets',
      rootNode: true,
    },
    sample: {
      cellIds: new Set([6, 7, 8]),
      name: 'Samples',
      key: 'sample',
      type: 'metadataCategorical',
      rootNode: true,
    },
    scratchpad: {
      cellIds: new Set(),
      name: 'Scratchpad',
      key: 'scratchpad',
      type: 'cellSets',
      rootNode: true,
    },
  },
};

export {
  // eslint-disable-next-line import/prefer-default-export
  mockCellSets,
};
