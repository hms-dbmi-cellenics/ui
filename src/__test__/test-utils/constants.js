export default {
  EXPERIMENT_ID: 'testae48e318dab9a1bd0bexperiment',
  CLONED_EXPERIMENT_ID: 'testae48e318dab9a1bd0bexperclone',
  PROJECT_WITH_SAMPLES_UUID: '380c53b4-mock-test-8091-4c82336d6d49',
  PROJECT_WITHOUT_SAMPLE_UUID: '4761594b-mock-test-ba89-c2c812d39bb5',
  EXPERIMENT_NAME: 'Test Experiment',
  PROJECT_ID: 'test24b6-8600-test-mock-63822d2fmock',
  SAMPLE_ID: 'test9188-d682-test-mock-cb6d644cmock',
  API_ENDPOINT: /^http?:\/\/localhost:3000.*$/,
  S3_ENDPOINT: 'http://mock.s3.amazonaws.com',
  // pragma: allowlist secret
  JWT_TOKEN: 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6Ikpfake.eyJzdWIiOiIxMjM0NTY3ODkwIiwifakeZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTfakeM5MDIyfQ.fakexwRfakeKKF2QT4fwpMeJf36POk6yJV_adQsfake',
  MOCK_DATETIME: '0000-00-00T00:00:00.000Z',
  ETAG: '380c53b4-mock-etag-8091-4c82336d6d49',
};
