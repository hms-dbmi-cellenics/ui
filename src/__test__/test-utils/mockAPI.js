import _ from 'lodash';

// Default platform data
import cellSetsData from '__test__/data/cell_sets.json';
import backendStatusData from '__test__/data/backend_status.json';
import processingConfigData from '__test__/data/processing_config.json';
import mockDemoExperiments from '__test__/test-utils/mockData/mockDemoExperiments.json';

import fake from '__test__/test-utils/constants';

import {
  responseData,
} from '__test__/test-utils/mockData';

const promiseResponse = (
  response,
  options = {},
) => Promise.resolve(new Response(response, options));

const statusResponse = (code, body) => (
  Promise.resolve({
    status: code,
    body: JSON.stringify(body),
  })
);

const delayedResponse = (response, delay = 10000, options = {}) => new Promise((resolve) => {
  setTimeout(() => {
    resolve(new Response(response, options));
  }, delay);
});

const workerDataResult = (data) => Promise.resolve(_.cloneDeep(data));

const fetchWorkMock = (
  mockedResults,
) => ((experimentId, body) => {
  if (typeof mockedResults[body.name] === 'function') {
    return mockedResults[body.name]();
  }

  return workerDataResult(mockedResults[experimentId]);
});

const generateDefaultMockAPIResponses = (experimentId) => ({
  [`experiments/${experimentId}$`]: () => promiseResponse(
    JSON.stringify(responseData.experiments.find(({ id }) => id === experimentId)),
  ),
  [`experiments/${experimentId}/processingConfig$`]: () => promiseResponse(
    JSON.stringify(processingConfigData),
  ),
  [`experiments/${experimentId}/cellSets$`]: () => promiseResponse(
    JSON.stringify(cellSetsData),
  ),
  [`experiments/${experimentId}/backendStatus$`]: () => promiseResponse(
    JSON.stringify(backendStatusData),
  ),
  experiments$: () => promiseResponse(
    JSON.stringify(responseData.experiments),
  ),
  [`experiments/${experimentId}/samples$`]: () => promiseResponse(
    JSON.stringify(responseData.samples[0]),
  ),
  '/v2/experiments/examples$': () => promiseResponse(
    JSON.stringify(mockDemoExperiments),
  ),
  'experiments/clone$': () => promiseResponse(
    JSON.stringify(fake.CLONED_EXPERIMENT_ID),
  ),
  [`/v2/workRequest/${experimentId}`]: () => statusResponse(200, 'OK'),
});

const mockAPI = (apiMapping) => (req) => {
  const path = req.url;

  const key = _.find(Object.keys(apiMapping), (matcherStr) => new RegExp(matcherStr).test(path));

  if (!key) {
    return statusResponse({
      status: 404,
      body: `Path ${path} is undefined`,
    });
  }

  return apiMapping[key](req);
};

export default mockAPI;
export {
  generateDefaultMockAPIResponses,
  promiseResponse,
  statusResponse,
  delayedResponse,
  workerDataResult,
  fetchWorkMock,
};
