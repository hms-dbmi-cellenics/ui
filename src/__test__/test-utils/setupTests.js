import '@testing-library/jest-dom/extend-expect';
import preloadAll from 'jest-next-dynamic';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('localforage');
jest.mock('utils/pushNotificationMessage');

// This is needed, because the unit tests don't register the Vitessce imports
jest.mock('components/data-exploration/DynamicVitessceWrappers', () => ({
  Scatterplot: () => 'Mocked Scatterplot',
  Heatmap: () => 'Mocked Heatmap',
  Spatial: () => 'Mocked Spatial',
}));

beforeAll(async () => {
  // Add stuff that needs to run once, before all tests
  await preloadAll();
});

beforeEach(async () => {
  // Add stuff that needs to run before each test
});
