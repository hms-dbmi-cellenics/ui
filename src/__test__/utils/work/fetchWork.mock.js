const mockGenesListData = ['A', 'B', 'C', 'D'];

// const mockGeneExpressionData = {
//   A: {
//     rawExpression: {
//       min: 0,
//       max: 6.8,
//       mean: 1.68,
//       stdev: 2.597331964,
//       expression: [0, 0, 0, 1.56, 0, 6.8, 3.4],
//     },
//     truncatedExpression: {
//       min: 0,
//       max: 6.8,
//       mean: 1.68,
//       stdev: 2.597331964,
//       expression: [0, 0, 0, 1.56, 0, 6.8, 3.4],
//     },
//   },
//   B: {
//     rawExpression: {
//       min: 0,
//       max: 6.8,
//       mean: 1.702857143,
//       stdev: 2.551115536,
//       expression: [0, 0, 0, 2.56, 0, 6.8, 2.56],
//     },
//     truncatedExpression: {
//       min: 0,
//       max: 6.8,
//       mean: 1.702857143,
//       stdev: 2.551115536,
//       expression: [0, 0, 0, 2.56, 0, 6.8, 2.56],
//     },
//   },
//   C: {
//     rawExpression: {
//       min: 0,
//       max: 3.4,
//       mean: 1.68,
//       stdev: 2.141525936,
//       expression: [0, 0, 0, 3.56, 0, 4.8, 3.4],
//     },
//     truncatedExpression: {
//       min: 0,
//       max: 3.4,
//       mean: 1.68,
//       stdev: 2.141525936,
//       expression: [0, 0, 0, 3.56, 0, 4.8, 3.4],
//     },
//   },
//   D: {
//     rawExpression: {
//       min: 0,
//       max: 6.8,
//       mean: 1.68,
//       stdev: 2.597331964,
//       expression: [0, 0, 0, 1.56, 0, 6.8, 3.4],
//     },
//     truncatedExpression: {
//       min: 0,
//       max: 6.8,
//       mean: 1.68,
//       stdev: 2.597331964,
//       expression: [0, 0, 0, 1.56, 0, 6.8, 3.4],
//     },
//   },
//   E: { hello: 'world' },
// };

// const mockCacheGet = jest.fn((x) => {
//   const mockCacheContents = {
//     fd3161a878f67ebf54018720cffd6a66: 'A', // pragma: allowlist secret
//     '10250a11679234110a1c260d6fd81d3c': 'B', // pragma: allowlist secret
//     f5c957411a28de68f35e1f5c8a29da7e: 'C', // pragma: allowlist secret
//     '33faa711a94a2028b5bae1778126aec0': 'E', // pragma: allowlist secret
//   };

//   if (x in mockCacheContents) {
//     return mockGeneExpressionData[mockCacheContents[x]];
//   }

//   return null;
// });

const mockCacheGet = jest.fn();
const mockCacheSet = jest.fn();
const mockCacheRemove = jest.fn();

const mockFetchWorkRequest = jest.fn(() => true);

const mockSeekFromS3 = jest.fn();

const mockCacheModule = {
  get: jest.fn((x) => mockCacheGet(x)),
  set: jest.fn((key, val) => mockCacheSet(key, val)),
  _remove: jest.fn((key) => mockCacheRemove(key)),
};

const mockFetchWorkModule = {
  __esModule: true,
  fetchWork: mockFetchWorkRequest,
};

const mockQcPipelineStartDate = '2021-01-01T01:01:01.000Z';

export {
  mockGenesListData, mockCacheGet, mockCacheSet,
  mockFetchWorkRequest, mockSeekFromS3,
  mockQcPipelineStartDate, mockCacheModule, mockFetchWorkModule,
};
