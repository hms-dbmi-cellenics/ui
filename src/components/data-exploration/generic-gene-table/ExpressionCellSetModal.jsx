/* eslint-disable react/no-array-index-key */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import {
  Modal, Form, Button, Select, InputNumber, Row, Alert, Col,
} from 'antd';

import createCellSetByExpression from 'redux/actions/cellSets/createCellSetByExpression';

const formItemStyle = { margin: '0.375rem 0' };
const FORM_NAME = 'filterForm';

const ExpressionCellSetModal = (props) => {
  const { onCancel } = props;

  const [isCreatingCellSet, setIsCreatingCellSet] = useState(false);

  const selectedGenes = useSelector((state) => state.genes.selected);
  const { experimentId } = useSelector((state) => state.experimentSettings.info);

  const dispatch = useDispatch();

  const [form] = Form.useForm();

  const comparisonOptions = [{
    value: 'greaterThan',
    label: 'Greater than',
  }, {
    value: 'lessThan',
    label: 'Less than',
  },
  ];

  const intialFormValues = selectedGenes.map((gene) => ({
    geneName: gene,
    comparisonType: comparisonOptions[0].value,
    thresholdValue: 0,
  }));

  const createExpressionCellSet = async () => {
    const formValues = form.getFieldValue(FORM_NAME);
    setIsCreatingCellSet(true);
    try {
      await dispatch(createCellSetByExpression(experimentId, formValues));
      setIsCreatingCellSet(false);
      onCancel();
    } catch (e) {
      setIsCreatingCellSet(false);
    }
  };

  return (
    <Modal
      open
      title='Create a new cell set based on gene expression'
      onCancel={onCancel}
      footer={[
        <Button
          key='createExpressionCellSet'
          type='primary'
          loading={isCreatingCellSet}
          disabled={isCreatingCellSet}
          onClick={createExpressionCellSet}
        >
          {isCreatingCellSet ? 'Creating cell set...' : 'Create'}
        </Button>,
      ]}
    >
      <Form
        form={form}
        size='middle'
        style={{ marginBottom: '1rem' }}
        initialValues={{
          geneName: '',
          comparisonType: 'greaterThan',
          thresholdValue: 0,
        }}
      >
        <Form.List
          name={FORM_NAME}
          initialValue={intialFormValues}
        >
          {(fields) => (
            <div style={{ maxHeight: '60vh', overflowY: 'auto', overflowX: 'hidden' }}>
              {fields.map((field, index) => {
                const { geneName } = form.getFieldValue(FORM_NAME)[index];
                return (
                  <Row gutter={8} key={`geneComparison-${index}`}>
                    <Col flex='auto'>
                      <Form.Item
                        name={[field.name, 'geneName']}
                        style={formItemStyle}
                      >
                        <strong>{geneName}</strong>
                      </Form.Item>
                    </Col>
                    <Col>
                      <Form.Item
                        name={[field.name, 'comparisonType']}
                        style={{
                          ...formItemStyle,
                          width: '130px',
                        }}
                      >
                        <Select
                          placeholder='Select comparison'
                          options={comparisonOptions}
                        />
                      </Form.Item>
                    </Col>
                    <Col>
                      <Form.Item
                        name={[field.name, 'thresholdValue']}
                        style={{
                          ...formItemStyle,
                          width: '80px',
                        }}
                      >
                        <InputNumber
                          step={0.01}
                          min={0}
                          placeholder='Insert threshold value'
                          style={{ width: '100%' }}
                        />
                      </Form.Item>
                    </Col>
                  </Row>
                );
              })}
            </div>
          )}
        </Form.List>
      </Form>
      <Alert type='info' message='To edit the list of genes, return to the gene list and change your selection.' style={{ margin: '0.5em 0' }} />
      <Alert
        type='warning'
        message={(
          <>
            Due to
            {' '}
            <a href='https://www.nature.com/articles/s41467-020-14976-9' target='_blank' rel='noreferrer'>high dropout</a>
            {' '}
            in scRNA-seq, cell sets defined using only a few genes are potentially misleading. We recommend using these cell sets for exploratory purposes only and to use computed clusters for comparisons.
          </>
        )}
      />
    </Modal>
  );
};

ExpressionCellSetModal.propTypes = {
  onCancel: PropTypes.func.isRequired,
};

export default ExpressionCellSetModal;
