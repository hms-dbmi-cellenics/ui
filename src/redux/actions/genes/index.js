import loadGeneExpression from './loadGeneExpression';
import changeGeneSelection from './changeGeneSelection';
import loadPaginatedGeneProperties from './loadPaginatedGeneProperties';
import loadMarkerGenes from './loadMarkerGenes';
import loadDownsampledGeneExpression from './loadDownsampledGeneExpression';

export {
  loadGeneExpression,
  changeGeneSelection,
  loadPaginatedGeneProperties,
  loadMarkerGenes,
  loadDownsampledGeneExpression,
};
