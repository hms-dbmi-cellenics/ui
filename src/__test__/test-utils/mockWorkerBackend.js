import * as socketConnectionMocks from 'utils/socketConnection';
import * as fetchWorkResponseMocks from 'utils/work/fetchWork';

import SocketMock from 'socket.io-mock';

const socketMock = new SocketMock();

jest.mock('utils/work/fetchWork', () => {
  const mockSeekFromS3 = jest.fn();
  const originalModule = jest.requireActual('utils/work/fetchWork');

  return {
    __esModule: true, // Use it when dealing with esModules
    ...originalModule,
    fetchWork: mockSeekFromS3,
  };
});

jest.mock('aws-amplify', () => ({
  configure: () => ({
    Storage: {
      AWSS3: {
        bucket: 'biomage-originals-test',
      },
    },
  }),
  Storage: {
    get: async (ETag) => `http://mock.s3.amazonaws.com/${ETag}`,
  },
  Auth: {
    federatedSignIn: jest.fn(),
    signOut: jest.fn(),
    currentAuthenticatedUser: jest.fn().mockImplementation(async () => ({
      username: 'mockuser',
      attributes: {
        email: 'mock@user.name',
        name: 'Mocked User',
      },
    })),
    currentSession: () => ({
      getIdToken: (() => ({
        getJwtToken: () => 'fakeJwtToken',
      })),
    }),
  },
  Hub: {
    listen: jest.fn(),
  },
}));

jest.mock('utils/socketConnection', () => {
  const mockEmit = jest.fn();
  const mockOn = jest.fn();
  return {
    __esModule: true,
    default: new Promise((resolve) => {
      resolve({ emit: mockEmit, on: mockOn, id: '1234' });
    }),
    mockEmit,
    mockOn,
  };
});

// To mock worker response, modify the response returned from the API
// Set to null to force fetch from API
fetchWorkResponseMocks.fetchWork.mockImplementation(() => null);

// Set up socket emitter mock
socketConnectionMocks.mockEmit.mockImplementation((workRequestType, requestBody) => {
  const responseBody = { response: { error: false } };

  // After emitting, send reply to listener
  socketMock.socketClient.emit(`WorkResponse-${requestBody.ETag}`, responseBody);
});

// Set up socket listener mock
socketConnectionMocks.mockOn.mockImplementation((channel, f) => {
  socketMock.on(channel, (responseBody) => f(responseBody));
});
