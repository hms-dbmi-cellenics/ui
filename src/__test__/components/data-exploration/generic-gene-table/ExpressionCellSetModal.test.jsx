import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
import { makeStore } from 'redux/store';

import ExpressionCellSetModal from 'components/data-exploration/generic-gene-table/ExpressionCellSetModal';

import { GENES_SELECT } from 'redux/actionTypes/genes';
import { dispatchWorkRequest } from 'utils/work/seekWorkResponse';
import { loadBackendStatus } from 'redux/actions/backendStatus';
import { updateExperimentInfo } from 'redux/actions/experimentSettings';

import mockAPI, { generateDefaultMockAPIResponses } from '__test__/test-utils/mockAPI';
import fake from '__test__/test-utils/constants';
import endUserMessages from 'utils/endUserMessages';

enableFetchMocks();

const mockOnCancel = jest.fn();

jest.mock('utils/work/seekWorkResponse', () => ({
  seekFromS3: jest.fn(),
  dispatchWorkRequest: jest.fn(),
}));

const renderExpressionCellSetModal = async (storeState) => {
  await act(async () => {
    render(
      <Provider store={storeState}>
        <ExpressionCellSetModal onCancel={mockOnCancel} />
      </Provider>,
    );
  });
};

let storeState;
const selectedGenes = ['DOK3', 'DOK4'];
const experimentId = fake.EXPERIMENT_ID;

const createButtonText = /^Create$/i;
const loadingText = /Creating cell set.../i;

describe('ExpressionCellSetModal', () => {
  beforeEach(async () => {
    jest.clearAllMocks();

    storeState = makeStore();

    fetchMock.resetMocks();
    fetchMock.mockIf(/.*/, mockAPI(generateDefaultMockAPIResponses(experimentId)));

    await storeState.dispatch({
      type: GENES_SELECT,
      payload: {
        experimentId,
        genes: selectedGenes,
      },
    });
    await storeState.dispatch(loadBackendStatus(experimentId));
    await storeState.dispatch(updateExperimentInfo({ experimentId }));
  });

  it('renders correctly', async () => {
    await renderExpressionCellSetModal(storeState);

    expect(screen.getAllByText('Create a new cell set based on gene expression')).toHaveLength(1);
    expect(screen.getByText('DOK3')).toBeInTheDocument();
    expect(screen.getByText('DOK4')).toBeInTheDocument();
    expect(screen.getByText('Create')).toBeInTheDocument();
    expect(screen.getAllByText('Greater than')).toHaveLength(2);
    expect(screen.getAllByRole('spinbutton')).toHaveLength(2);
    expect(screen.getAllByLabelText('close')[0]).toBeInTheDocument();
  });

  it('Sends the correct genes list with params', async () => {
    await renderExpressionCellSetModal(storeState);

    expect(screen.getByText(createButtonText).closest('button')).not.toBeDisabled();

    // Change the dropdown to less than
    act(() => {
      userEvent.click(screen.getAllByText('Greater than')[0]);
    });

    userEvent.click(screen.getByText('Less than'), undefined, { skipPointerEventsCheck: true });

    // Input another threshold value
    userEvent.type(screen.getAllByRole('spinbutton')[0], '{backspace}5');

    await act(async () => {
      userEvent.click(screen.getByText(createButtonText));
    });

    const requestParams = dispatchWorkRequest.mock.calls[0];
    expect(requestParams).toMatchSnapshot();
  });

  it('Modal closes on success', async () => {
    dispatchWorkRequest.mockImplementationOnce(() => new Promise((resolve) => {
      setTimeout(resolve(null), 2000);
    }));

    await renderExpressionCellSetModal(storeState);

    await act(async () => {
      userEvent.click(screen.getByText(createButtonText).closest('button'));
    });

    waitFor(() => {
      expect(screen.getByText(loadingText)).toBeInTheDocument();
      expect(screen.getByText(loadingText).closest('button')).toBeDisabled();
    });

    expect(screen.getByText(createButtonText)).not.toBeDisabled();
    expect(screen.queryByText(loadingText)).toBeNull();

    expect(mockOnCancel).toHaveBeenCalledTimes(1);
  });

  it('Modal shows error notification and does not close on error', async () => {
    dispatchWorkRequest.mockImplementationOnce(() => Promise.reject(new Error('Some error')));

    await renderExpressionCellSetModal(storeState);

    await act(async () => {
      await userEvent.click(screen.getByText(createButtonText).closest('button'));
    });

    waitFor(() => {
      expect(screen.getByText(endUserMessages.ERROR_FETCHING_CELL_SETS)).toBeInTheDocument();
    });

    expect(mockOnCancel).not.toHaveBeenCalled();
  });
});
