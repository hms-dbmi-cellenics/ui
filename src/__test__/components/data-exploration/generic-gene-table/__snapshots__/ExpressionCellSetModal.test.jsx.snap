// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpressionCellSetModal Sends the correct genes list with params 1`] = `
Array [
  "testae48e318dab9a1bd0bexperiment",
  Object {
    "genesConfig": Array [
      Object {
        "comparisonType": "lessThan",
        "geneName": "DOK3",
        "thresholdValue": 0.05,
      },
      Object {
        "comparisonType": "greaterThan",
        "geneName": "DOK4",
        "thresholdValue": 0,
      },
    ],
    "name": "GetExpressionCellSets",
  },
  180,
  "bff42d1a0fd631bdb2a72de335468764",
  Object {
    "PipelineRunETag": "2021-10-20T12:51:44.755Z",
    "broadcast": true,
  },
]
`;
